### DISCUSSION TITLE
Enhance header search in files

### DISCUSSION THREAD

**Discussion by tadashi-aikawa** [tadashi-aikawa/obsidian-another-quick-switcher#41] [Ideas]
Just idea 💡  want to scroll when the focus moves. (maybe it is difficult without close the quick switcher...)

https://user-images.githubusercontent.com/9500018/167292121-0bd8a535-c9f7-423e-8f2f-68733eb1383a.mp4

![image](https://user-images.githubusercontent.com/9500018/167292146-d5bbb777-8a7c-4680-a2e4-c5e6b7f26081.png)

![image](https://user-images.githubusercontent.com/9500018/167292158-17e27919-86f8-4c40-baf3-1d1d707bc44d.png)

![image](https://user-images.githubusercontent.com/9500018/167292165-e0e39cd7-bcad-42e2-93af-a438a13b87fc.png)

**Comment by FelipeRearden** [comment]
> Just idea 💡  want to scroll when the focus moves. 

Wow Wow Wow that's a fantastic idea @tadashi-aikawa !!!!
- A floating outliner viewer with search and keyboard support !!!!!!!

> (maybe it is difficult without close the quick switcher...)

100% agree about keep the modal open while focus on the matches in the modal 

Thanks for sharing this idea :)

**Comment by tadashi-aikawa** [comment]
Fortunately, it would be easy than I had expected 😃 

https://user-images.githubusercontent.com/9500018/167307631-31d9dc32-f40d-4b7e-b3b4-edfeb3711911.mp4

**Reply by FelipeRearden** [reply]
AMAZING @tadashi-aikawa !!!!! 👀 👀 👀 👀 👀

I don't know what is your idea about how to present this new modal but I have two insights about this ....

Just to be easier for me two explain I'm gonna use these two names/terms:
- Heading Search HS (#28)
- Heading Floating Search HFS

1️⃣ A New Command to invoke  HFS as an unique modal

2️⃣ A new hotkey on the modal inside both HS and HFS to **switch** between the two modes Standard and Floating 

Just ideas for us to talk :)

**Reply by tadashi-aikawa** [reply]
> Heading Floating Search HFS

That is a good name 👍 

> 1️⃣ A New Command to invoke HFS as an unique modal

Yes. I am going to add the above command.

> 2️⃣ A new hotkey on the modal inside both HS and HFS to switch between the two modes Standard and Floating

It will be possible, but I don't know if that is necessary 🤔

**Reply by FelipeRearden** [reply]
> > Heading Floating Search HFS
> 
> 
> 
> That is a good name 👍 

💙

> > 1️⃣ A New Command to invoke HFS as an unique modal
> 
> 
> 
> Yes. I am going to add the above command.

🎉

> > 2️⃣ A new hotkey on the modal inside both HS and HFS to switch between the two modes Standard and Floating
> 
> 
> 
> It will be possible, but I don't know if that is necessary 🤔 

Yeah, I agree with you I think I overthink a little bit about this one :)


This is amazing @tadashi-aikawa really amazing 👀

**Comment by tadashi-aikawa** [comment]
Released [4.12.0-beta1](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.12.0-beta1) 🚀

**Comment by FelipeRearden** [comment]
> Released [4.12.0-beta1](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.12.0-beta1) 🚀 

Starting tests right now ... I get back to you in less than 30 minutes 🚀

**Reply by FelipeRearden** [reply]
@tadashi-aikawa 

Working absolutely perfect on version `4.12-beta1` on Obsidian  `0.14.6`

## Idea 💡

I have one idea to share with you that is related to position of the active note when HFS is ON

**Idea** would be possible to to have the command `Header Floating Search in File` as **Toggle**?
- _When I hit the command HFS will be ON and if I hit again HFS will be OFF_

Let me share why 🧐 

## Workflow 

**Resources**
- AQS plugin 💙
- [QuickAdd plugin](https://github.com/chhoumann/quickadd) 
- [Snippets Commands plugin](https://github.com/deathau/snippet-commands-obsidian)
- Hotkeys++ plugin

**Idea**
When  HFS is ON I wanna reduce the width of the Active note to avoid HFS overflow the content of the note.

**Operation in Obsidian**

ℹ️ Create a Snippet that changes the width of the note when Readable Line Length is ON

![03](https://user-images.githubusercontent.com/85180696/167395803-8c6eeba8-a68d-4671-b47c-94162cfd8004.png)

ℹ️ Hotkeys++ has a command that lets you toggle the Readable Line Length

ℹ️ Snippets Commands creates a command for the snippets to toggle them ON or OFF

ℹ️ AQS HFS command as a toggle 

ℹ️ A QuickAdd MACRO command that lets me merge all of these commands into a one command 

ℹ️ Define a a hotkey for the QuickAdd Macro command 

**Final Result**

When I hit the hotkey it will open HFS and position the active note in the left 

![02](https://user-images.githubusercontent.com/85180696/167395846-cdb54e48-68c8-400d-82ee-e03519f7f620.png)


When I hit the hotkey again it will close the HFS and get back the note to the normal position 

![01](https://user-images.githubusercontent.com/85180696/167395937-f83d81e4-604c-4567-b0dd-ed3a88fd4f99.png)


---

Lest me know what you think :)

**Reply by tadashi-aikawa** [reply]
@FelipeRearden 
Thank you for your testing and an idea! 😄 

> Idea would be possible to to have the command Header Floating Search in File as Toggle?

I think it seems subtle. The reason is that we can't execute commands while a quick switcher opens.

By any chance, A QuickAdd MACRO can execute commands while a quick switcher opens? 🤔

**Reply by FelipeRearden** [reply]
> Thank you for your testing and an idea! 😄

💙

> By any chance, A QuickAdd MACRO can execute commands while a quick switcher opens? 🤔

I can confirm that is possible to run commands after the quick switcher is open :)

I even have an FR to add a WAIT command to avoid problems using commands after modals are open  
🔗 https://github.com/chhoumann/quickadd/issues/118

---

In this specific case, I would built the MACRO using the HFS command in the **last**. I‘m thinking about something like this

🖼 MACRO commands order
![EB0CEB22-A6D2-43A9-A082-DBBA30DA7989](https://user-images.githubusercontent.com/85180696/167448920-cff272a0-ec35-4402-8371-45cfffdafe60.jpeg)

ℹ️ I will open the Modal only in the end of the Commands list :) I change the Active Note before opening HFS


🖼 Hotkey 
![DF632F41-74D4-4CC8-AA39-D6F9D0F37F17](https://user-images.githubusercontent.com/85180696/167449157-4bd69876-cc91-4aaf-8015-2c5e5e7cb045.jpeg)

---

Let me know if we are on the same page :)

**Reply by tadashi-aikawa** [reply]
Thank you for details!

I understand why and what you want to do, but I am sorry, I do not want to add it, at least not at this time, because it would not benefit most users and would be confusing.

Kind regards 🙏

**Reply by FelipeRearden** [reply]
> I understand why and what you want to do, but I am sorry, I do not want to add it, at least not at this time, because it would not benefit most users and would be confusing.

It's 100% okay @tadashi-aikawa !!!

Thank your for kindness and your attention to my idea :)

We already have a fantastic feature to use :)

Have a great day and let's keep talking about AQS and VC ☀️

**Comment by tadashi-aikawa** [comment]
Released [v4.12.0-beta3](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/4.12.0-beta3) 🚀 

It is probably a final beta version if there are no critical problems.

https://user-images.githubusercontent.com/9500018/167628680-7c7e118d-c674-4d84-97d7-4aed025cae0b.mp4

**Reply by FelipeRearden** [reply]
Working perfect on version `4.12.beta3` on Obsidian `0.14.6`

Have a great day @tadashi-aikawa ☀️

**Reply by tadashi-aikawa** [reply]
Thanks!! I'll release v4.12.0 next morning. (about 2022-05-11T23:00 UTC)

**Reply by FelipeRearden** [reply]
Great @tadashi-aikawa !!!

See you tomorrow :)

**Reply by FelipeRearden** [reply]
Hello @tadashi-aikawa !!!!

Working perfect on version `4.12.0` Obsidian `0.14.6`

Thank you very much !!!!!!!!
